core/src/main/java/org/infinispan/interceptors/locking/NonTransactionalLockingInterceptor.java

```diff
@@ -53,19 +53,11 @@ public Object visitGetAllCommand(InvocationContext ctx, GetAllCommand command) t
    public Object visitPutMapCommand(InvocationContext ctx, PutMapCommand command) throws Throwable {
       assertNonTransactional(ctx);
       try {
-         if (!command.isForwarded()) {
-            boolean skipLocking = hasSkipLocking(command);
-            long lockTimeout = getLockAcquisitionTimeout(command, skipLocking);
-            for (Object key : command.getMap().keySet()) {
-               if (shouldLock(key, command))
-                  lockKey(ctx, key, lockTimeout, skipLocking);
-            }
+         if (!command.isForwarded() && !hasSkipLocking(command)) {
+            lockAllAndRecord(ctx, command.getMap().keySet().stream().filter(this::shouldLockKey), getLockTimeoutMillis(command));
```

Comment:
Ah, back when I was trying those changes you didn't have the `lockAndRecord()` overload that took a collection.